src/VisualStudio/CSharp/Impl/Options/Formatting/CodeStylePage.cs

```diff
@@ -11,7 +18,422 @@ internal class CodeStylePage : AbstractOptionPage
+                {
+                    if (ruleString.Length != 0)
+                    {
+                        ruleString.Append(",");
+                    }
+
+                    ++rulesApplied;
+                    ruleString.Append("anonymous_methods");
+                }
+                if (optionSet.GetOption(CSharpFormattingOptions.NewLinesForBracesInAnonymousTypes))
+                {
+                    if (ruleString.Length != 0)
+                    {
+                        ruleString.Append(",");
+                    }
+
+                    ++rulesApplied;
+                    ruleString.Append("anonymous_types");
+                }
+                if (optionSet.GetOption(CSharpFormattingOptions.NewLinesForBracesInControlBlocks))
+                {
+                    if (ruleString.Length != 0)
+                    {
+                        ruleString.Append(",");
+                    }
+
+                    ++rulesApplied;
+                    ruleString.Append("control_blocks");
+                }
+                if (optionSet.GetOption(CSharpFormattingOptions.NewLinesForBracesInLambdaExpressionBody))
+                {
+                    if (ruleString.Length != 0)
+                    {
+                        ruleString.Append(",");
+                    }
+
+                    ++rulesApplied;
+                    ruleString.Append("lambdas");
+                }
+                if (optionSet.GetOption(CSharpFormattingOptions.NewLinesForBracesInMethods))
+                {
+                    if (ruleString.Length != 0)
+                    {
+                        ruleString.Append(",");
+                    }
+
+                    ++rulesApplied;
+                    ruleString.Append("methods");
+                }
+                if (optionSet.GetOption(CSharpFormattingOptions.NewLinesForBracesInObjectCollectionArrayInitializers))
+                {
+                    if (ruleString.Length != 0)
+                    {
+                        ruleString.Append(",");
+                    }
+
+                    ++rulesApplied;
+                    ruleString.Append("object_collection");
+                }
+                if (optionSet.GetOption(CSharpFormattingOptions.NewLinesForBracesInProperties))
+                {
+                    if (ruleString.Length != 0)
+                    {
+                        ruleString.Append(",");
+                    }
+
+                    ++rulesApplied;
+                    ruleString.Append("properties");
+                }
+                if (optionSet.GetOption(CSharpFormattingOptions.NewLinesForBracesInTypes))
+                {
+                    if (ruleString.Length != 0)
+                    {
+                        ruleString.Append(",");
+                    }
+
+                    ++rulesApplied;
+                    ruleString.Append("types");
+                }
```

Comment:
can you try refactoring this? maybe storing these pairs as a dictionary or a list of tuples and then looping over them? and then use the length of that instead of hardcoding "9"?